### PR TITLE
add partition metadata only save functionality

### PR DESCRIPTION
**Pull request by zhljen**
- add partition metadata only save functionality
- remove unused metric variables

### CODE REVIEW

File: metacat-common-server/src/main/java/com/netflix/metacat/common/server/events/MetacatSaveTablePartitionMetadataOnlyPreEvent.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+/**
+ * Metacat save table partition metadata only pre event.
+ * @author zhenl
+ * @since 1.2.0
+ */
+@Getter
+@EqualsAndHashCode(callSuper = true)
+@ToString(callSuper = true)
+public class MetacatSaveTablePartitionMetadataOnlyPreEvent extends MetacatSaveTablePartitionPreEvent {
```
Comment (ajoymajumdar) [inline metacat-common-server/src/main/java/com/netflix/metacat/common/server/events/MetacatSaveTablePartitionMetadataOnlyPreEvent.java]: When extending MetacatSaveTablePartitionPreEvent, will the listeners listening on MetacatSaveTablePartitionPreEvent get called for a MetacatSaveTablePartitionMetadataOnlyPreEvent?

File: metacat-functional-tests/src/functionalTest/groovy/com/netflix/metacat/MetacatFunctionalSpec.groovy [hunk 2]
```diff
@@ -923,6 +906,69 @@ class MetacatFunctionalSpec extends Specification {
...
+        then:
+        response.updated.empty
+        response.added.empty
+
+        when:
+        allPartitions = partitionApi.getPartitions(name.catalogName, name.databaseName, name.tableName, null, null, null, null, null, true)
+        savedPartition = allPartitions?.find { it.name == escapedName }
+
+        then:
+        savedPartition.definitionMetadata.get('part_def_field').longValue() == -1
```
Comment (ajoymajumdar) [inline metacat-functional-tests/src/functionalTest/groovy/com/netflix/metacat/MetacatFunctionalSpec.groovy]: How are you making sure that the partition info is not getting updated?

File: metacat-common-server/src/main/java/com/netflix/metacat/common/server/monitoring/Metrics.java [hunk 3]
```diff
@@ -40,16 +40,10 @@
     CounterThrift(Component.thrift, Type.counter, "request"),
 
     /**
-     * Gauge.
-     */
```
[the remark this answers is not in the document]
Comment (zhljen) [reply metacat-common-server/src/main/java/com/netflix/metacat/common/server/monitoring/Metrics.java]: No, we replace them with distribution summary already. I remove the unused code here.

File: metacat-common-server/src/main/java/com/netflix/metacat/common/server/events/MetacatSaveTablePartitionMetadataOnlyPreEvent.java (hunk 1 above)
Comment (zhljen) [reply metacat-common-server/src/main/java/com/netflix/metacat/common/server/events/MetacatSaveTablePartitionMetadataOnlyPreEvent.java]: Will change to separate them out.

File: metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java [hunk 4]
```diff
@@ -276,12 +276,38 @@ public PartitionsSaveResponseDto save(final QualifiedName name, final Partitions
     }
 
     /**
+     * optimization for metadata only updates
```
Comment (ajoymajumdar) [inline metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java]: Add a period at the end of the sentence. Checkstyle is failing in the build..

File: metacat-functional-tests/src/functionalTest/groovy/com/netflix/metacat/MetacatSmokeSpec.groovy [hunk 5]
```diff
@@ -478,6 +481,7 @@ class MetacatSmokeSpec extends Specification {
             if (repeat) {
                 assert api.getTable(catalogName, databaseName, tableName, false, true, false).getDefinitionMetadata().get('savePartitions') != null
                 assert partitions.get(0).dataUri == uri + '0' + uriResult
+                assert partitions.get(0).getDataMetadata().get('updateMetadata') != null
```
Comment (ajoymajumdar) [inline metacat-functional-tests/src/functionalTest/groovy/com/netflix/metacat/MetacatSmokeSpec.groovy]: How is this testing this feature? You should get the partition creation date before saving and validate that it does not change after saving with the saveMetadataOnly flag set to true.

File: metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java [hunk 6]
```diff
@@ -216,28 +215,75 @@ public Integer count(final QualifiedName name) {
...
-        registry.distributionSummary(
-            this.partitionAddDistSummary.withTags(name.parts())).record(partitionDtos.size());
-
         if (!tableService.exists(name)) {
             throw new TableNotFoundException(name);
         }
+        //optimization for metadata only updates (e.g. squirrel) , assuming only validate partitions are requested
+        if (dto.getSaveMetadataOnly()) {
+            return savePartitionMetadataOnly(metacatRequestContext, dto, name, partitionDtos);
+        }
```
Comment (ajoymajumdar) [inline metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java]: Can you do an else to make a clear separation.

Comment (zhljen) [reply metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java]: Sure. will do.

**Pull request merged**